### PR TITLE
Add model loading

### PR DESCRIPTION
**Pull request by Rohan138**

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -22,3 +25,15 @@ def forward(self, x):
         x = x.view(-1, 784)
         x = self.layers(x)
         return F.log_softmax(x, dim=1)
+
+class HuggingFaceModel:
+
+    def __init__(self, model_name: str = "zephyr/zephyr-7b-beta") -> None:
+        if os.path.exists(f"/data/public_models/{model_name}"):
```
Comment (mukobi) [inline models.py:32]: Consider adding a comment that this is specific to the CAIS cluster

File: models.py [hunk 2]
```diff
@@ -22,3 +25,15 @@ def forward(self, x):
         x = x.view(-1, 784)
         x = self.layers(x)
         return F.log_softmax(x, dim=1)
+
+class HuggingFaceModel:
```
Comment (mukobi) [inline models.py:29]: Does this need to inherit from torch.nn.Module for the `forward` override to work?

File: models.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
+import os
 from torch import nn
 from torch.nn import functional as F
 
+from transformers import AutoModel, AutoTokenizer
```
Comment (ShinnosukeUesaka) [inline models.py:5]: Some models are not supported by AutoModel.
Safer to use AutoModelForCausalLM.

**Review by ShinnosukeUesaka — CHANGES_REQUESTED** [review]

File: models.py (hunk 1 above)
Comment (Rohan138) [reply models.py]: Added comment and logger.info call

File: models.py (hunk 2 above)
Comment (Rohan138) [reply models.py]: Switched to `__call__` instead of forward

**Comment by Rohan138** [comment]
@mukobi could you review? thanks

File: main.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ def main(cfg: DictConfig):
     utils.check_base_results_saved(wandb.config.model.path, wandb.config.unadapt)
 
     logger.info("Loading model")
-    model_base = models.load_model(wandb.config.model, logger)
+    model_base = models.load_model(wandb.config.model, wandb.config.device)
```
Comment (mukobi) [inline main.py:34]: Could 'device' be stored under model in the config, like `wandb.config.model.device`? Seems cleaner that way. Also, good to pass in the logger so we can log issues or special notes about loading the model.

File: src/ufm/models.py [hunk 5]
```diff
@@ -26,14 +28,23 @@ def forward(self, x):
         x = self.layers(x)
         return F.log_softmax(x, dim=1)
 
-class HuggingFaceModel:
+class HuggingFaceModel():
```
Comment (mukobi) [inline src/ufm/models.py:31]: Why not inherit from `nn.Module`?

**Comment by Rohan138** [comment]
- Switched device to be under `model.config`
- Swapped back to using `__call__` instead of `nn.Module` to avoid ambiguity with moving this to a specific device, since we should be using `device_map=auto` for the model for parallelism
- Swapped paths to reflect new CAIS cluster path

**Pull request merged**